web/modules/custom/area_summary/src/Entity/AreaSummary.php

```diff
@@ -0,0 +1,110 @@
+ *     "edit-form" = "/area_summary/{area_summary}/edit",
+ *     "delete-form" = "/area_summary/area_summary/{area_summary}/delete",
+ *     "collection" = "/admin/content/area_summary",
+ *   },
+ * )
+ */
+class AreaSummary extends ContentEntityBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
+
+    $fields[$entity_type->getKey('id')] = BaseFieldDefinition::create('integer')
+      ->setLabel(new TranslatableMarkup('ID'))
+      ->setDescription(new TranslatableMarkup('The ID of the area summary entity.'))
+      ->setReadOnly(TRUE)
+      ->setSetting('unsigned', TRUE);
+
+    $fields[$entity_type->getKey('uuid')] = BaseFieldDefinition::create('uuid')
+      ->setLabel(new TranslatableMarkup('UUID'))
+      ->setDescription(new TranslatableMarkup('The UUID of the area summary entity.'))
+      ->setReadOnly(TRUE);
+
+    $fields['area'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Area'))
+      ->setDescription(t('The area aka offer/license.'))
+      ->setSetting('target_type', 'node')
+      ->setSetting('handler_settings', ['target_bundles' => ['offer_license' => 'offer_license']])
+      ->setDisplayOptions('form', ['weight' => 0]);
+
+    $fields['farmer'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Farmer'))
+      ->setDescription(t('The farmer.'))
+      ->setSetting('target_type', 'node')
+      ->setSetting('handler_settings', ['target_bundles' => ['farmer_details' => 'farmer_details']])
+      ->setDisplayOptions('form', ['weight' => 0]);
+
+    $fields['management_unit'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Management unit'))
+      ->setDescription(t('The management unit aka range.'))
+      ->setSetting('target_type', 'taxonomy_term')
+      ->setSetting('handler_settings', ['target_bundles' => ['management_unit' => 'management_unit']])
+      ->setDisplayOptions('form', ['weight' => 0]);
+
+    $fields['cfr'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('CFR'))
+      ->setDescription(t('The central forest reserve aka CFR.'))
+      ->setSetting('target_type', 'taxonomy_term')
+      ->setSetting('handler_settings', ['target_bundles' => ['central_forest_reserve' => 'central_forest_reserve']])
+      ->setDisplayOptions('form', ['weight' => 0]);
+
+    $fields['area_allocated'] = BaseFieldDefinition::create('decimal')
+      ->setLabel(new TranslatableMarkup('Total area allocated'))
+      ->setRevisionable(TRUE)
+      ->setDisplayOptions('form', ['weight' => 0]);
+
+    $fields['area_planted'] = BaseFieldDefinition::create('decimal')
+      ->setLabel(new TranslatableMarkup('Total area planted'))
+      ->setRevisionable(TRUE)
+      ->setDisplayOptions('form', ['weight' => 0]);
+
+    $fields['average_stems'] = BaseFieldDefinition::create('decimal')
+      ->setLabel(new TranslatableMarkup('Average stems/ha'))
```

Comment:
This is very interesting, I normally use a simple t()

Reply:
Simple t() is better :)